dango/lending/src/execute.rs

```diff
@@ -150,3 +151,30 @@ pub fn borrow(ctx: MutableCtx, coins: Coins) -> anyhow::Result<Response> {
     // Transfer the coins to the caller
     Ok(Response::new().add_message(Message::transfer(ctx.sender, coins)?))
 }
+
+pub fn repay(ctx: MutableCtx) -> anyhow::Result<Response> {
+    // Ensure all sent coins are whitelisted
+    for coin in &ctx.funds {
+        ensure!(
+            MARKETS.has(ctx.storage, coin.denom),
+            "Invalid denom. Only whitelisted denoms can be repaid."
+        );
+    }
+
+    let mut msgs = vec![];
```

Comment:
Instead of a `Vec<Message>`, you can create a

```rust
let mut maybe_msg = None;
```

and then

```rust
Ok(Response::new().may_add_message(maybe_msg))
```